README.md

```diff
@@ -1,2 +1,115 @@
+
+## Configurations
+
+### deploy-hosts.yaml
+
+This file defines the different hosts you may want to deploy to. Adjust those hosts to your needs.
+See [deployer documentation](https://deployer.org/docs/hosts) for further information.
+
+### deploy.php
+
+This file defines the basic configuration via deployers `set`-method. Make sure to adjust the following options to your needs:
+* `repository`: URL to the repository you want to deploy.
+* `bin/php`: Absolute path to the php executable on your server.
+* `bin/composer`: Path to composer. Adjust if you do not want to use the composer.phar.
+* `shopware_download_path`: URL to the shopware installations package you want to deploy.
+* `shopware_update_version`: Shopware version you want to update to. Will be used to check wether the current installations needs to be updated.
+* `shopware_update_path`: URL to shopware update package you want to update to.
+* `plugins`: An array of plugins that will be installed and updated during deployment. Those plugins need to be present in the codebase.
+* `theme_config`: An array of theme configuration settings that will be set during deployment.
+* `fcgi_sockets`: An array of fcgi_sockets that will be used to reset the OPCache.
+
+## kellerkinder tasks (deploy-shopware.php)
+
+### kellerkinder:shopware:install
+
+Wrapper task for the basic installations steps:
+
+#### kellerkinder:shopware:download
+Downloads the defined shopware installation package, unpacks it and rsyncs it into the shopware root directory.
+
+#### kellerkinder:vendors
+Requires some plugins via composer. In this example config those will be [K10rStaging](https://github.com/kellerkinderDE/K10rStaging) and [K10rDeployment](https://github.com/kellerkinderDE/K10rDeployment)
+
+_Note:_ They will not be installed yet.
+
+### kellerkinder:shopware:update
+This task will check wether the currently installed shopware version matches the `shopware_update_version` (see above) and will download the shopware update package and update the shop if needed.
+
+### kellerkinder:shopware:configure
+This tasks wraps several configuration steps:
+
+#### kellerkinder:shopware:opcache
+This will reset the OPCache using `cachetool.phar` and the `fcgi_sockets` specified in the `deploy.php`
+
+#### kellerkinder:shopware:plugins
+This will (re)install `K10rDeployment` which gives us some useful commands for the deployment process. Afterwards it deactivates `SwagUpdate` since we do not want anyone to install an update via backend.
+And most importantly it installs and updates all plugins specified in the `plugins` variable in `deploy.php`.
+
+This task should also be used to set plugin configurations that should be the same on all stages like this:
+```
+run("cd {{shopware_public_path}} && {{bin/php}} bin/console sw:plugin:config:set SwagImportExport useCommaDecimal 0");
```

Comment:
Wouldn't it be preferable to add a custom task (via `after`) with custom plugin settings in `deploy.php` to keep the `deploy-shopware.php` pristine?

Reply:
I decided to move the plugin config to an array that can be adjusted via deploy.php. I prefer this solution over a custom task. What do you think?